ui/src/app/components/copy-modal.tsx

```diff
@@ -218,27 +276,57 @@ class CopyModalComponent extends React.Component<Props, State> {
     }
   }
 
-  setDestination(destination: Workspace) {
+  // OK to copy a notebook with a mismatch, but show a warning message
+  setNotebookCdrMismatchWarning(destination: Workspace, fromCdrVersionId: string) {
+    const warningMsg = `The selected destination workspace uses a different dataset version ` +
+        `(${this.cdrName(destination.cdrVersionId)}) than the current workspace (${this.cdrName(fromCdrVersionId)}). ` +
+        'Edits may be required to ensure your analysis is functional and accurate.';
+    this.setState({ cdrMismatch: warningMsg, destination: destination });
+  }
+
+  // not OK to copy a Concept Set with a mismatch.  Show an error message and prevent copy
+  setConceptSetCdrMismatchError(destination: Workspace, fromCdrVersionId: string) {
+    const errorMsg = `Can’t copy to that workspace. It uses a different dataset version ` +
+        `(${this.cdrName(destination.cdrVersionId)}) than the current workspace (${this.cdrName(fromCdrVersionId)}).`;
+    this.setState({ cdrMismatch: errorMsg, destination: null });
+  }
+
+  validateAndSetDestination(destination: Workspace) {
+    const {fromCdrVersionId, resourceType} = this.props;
+
     this.clearCopyError();
-    this.setState({destination: destination});
+
+    if (fromCdrVersionId === destination.cdrVersionId) {
+      this.setState({cdrMismatch: '', destination: destination});
+      return;
+    }
+
+    if (resourceType === ResourceType.NOTEBOOK) {
+      this.setNotebookCdrMismatchWarning(destination, fromCdrVersionId);
+    } else if (resourceType === ResourceType.CONCEPTSET) {
+      this.setConceptSetCdrMismatchError(destination, fromCdrVersionId);
+    }
```

Comment:
opt: consider using an `fp.cond` here - I believe it would eliminate the need for an early return and would tie together the 3 conditions as "one of these can occur"

Reply:
I attempted to do so, but this caused my e2e tests to fail.  This attempt is commit ea710b52279e4e34f07b67eaf3cdd761ddeead7b